### ISSUE TITLE
datadog_monitor always displays changes for thresholds

### ISSUE THREAD

**Issue by billputer** [hashicorp/terraform#9197]
### Terraform Version

0.7.4

### Affected Resource(s)

datadog_monitor

### Terraform Configuration Files

Almost identical to configuration in [datadog_monitor documentation](https://www.terraform.io/docs/providers/datadog/r/monitor.html)
```hcl
resource "datadog_monitor" "foo" {
  name = "Name for monitor foo"
  type = "metric alert"
  message = "Monitor triggered. Notify: @none"

  query = "avg(last_1h):avg:aws.ec2.cpu{environment:foo,host:foo} by {host} > 2"

  thresholds {
    ok = 0
    warning = 1
    critical = 2
  }
}
```

### Debug Output

https://gist.github.com/billputer/83622712603e127b61407df05cbc2929

### Expected Behavior

`terraform plan` shows no changes

### Actual Behavior
`terraform plan` shows the following changes:
```
~ datadog_monitor.foo
    thresholds.critical: "2.0" => "2"
    thresholds.ok:       "0.0" => "0"
    thresholds.warning:  "1.0" => "1"
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`

### Important Factoids

Didn't happen on Terraform pre-0.7.

**Comment by billputer** [comment]
I've done some research, and it looks like this bug is a result of #8324 .

The thresholds from the Datadog Go client are now being mutated in resourceDatadogMonitorRead, which is doing something funky with the types. For example, I've found that setting the threshold to 2.5 works as intended, but both 2.0 or 2 result in the output `"2.0" => "2"`.

I'll do some more investigation and see if I can come up with a PR.

**Comment by eredi93** [comment]
+1

**Comment by nyanshak** [comment]
I made PR #9373 to fix.

**Comment by catsby** [comment]
I merged #9466 which should address this. Please let me know if still hit it after upgrading to the next release, or a build from the `master` branch
